### PR TITLE
Fix release tag lifecycle

### PR DESCRIPTION
**Pull request by YunchuWang**
# Summary

## What changed?
- Removed all pre-merge Git tag creation and force-push behavior from Prepare Release.
- Kept package-scoped tag computation for branch naming, changelog baselines, summaries, and post-merge guidance.
- Added fail-safe instructions to tag the verified merged `main` commit without force, and required the official build to use that exact commit.
- Added an all-tags Azure Pipelines trigger to `eng/ci/code-mirror.yml` while preserving the existing `main` trigger, so each newly-created post-merge package tag uses the shared mirror template's dedicated tag path immediately.
- Added an operator gate requiring the tag-specific mirror run to resolve the internal tag to the exact merged commit and re-lock it before official build/release proceeds.

## Why is this change needed?
- Prepare Release could move a GitHub release tag across reruns before the release PR merged, while the ADO mirror correctly refused to overwrite its existing tag. Release tags must be immutable and created only after merge.
- Without a tag trigger, the shared template's safe tag path is unreachable: a new release tag waits for a later `main` push, whose normal branch mirror path pushes `--tags` and can fail on an existing mismatched destination tag.
- After this PR merges, the existing Azure Managed mismatch can be reconciled by re-emitting the canonical GitHub tag through the tag-triggered pipeline path.

## Issues / work items
- Resolves #
- Related #

---

# Project checklist
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `CHANGELOG.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] Breaking change? No
  - [ ] If yes:
    - Impact: N/A
    - Migration guidance: N/A

---

# AI-assisted code disclosure (required)

## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): GitHub Copilot
- AI-assisted areas/files: `.github/workflows/prepare-release.yaml`, `eng/ci/code-mirror.yml`, `doc/release_process.md`
- What you changed after AI output: N/A; the generated diff was reviewed and validated directly.

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

---

# Testing

## Automated tests
- Result: Passed — both workflow YAML files parse; Azure trigger assertions preserve `main` and include all tags; rendered Summary shell syntax/content checks enforce the exact mirrored-commit and re-lock gate; existing/unused remote-tag guard cases pass; exact target-tag filtering uses fixed-string whole-line matching; forbidden tag-mutation assertions, `git diff --check`, and `npm run lint` pass.

## Manual validation (only if runtime/behavior changed)
- Environment (OS, Node.js version, components): Windows; generated GitHub Actions summary rendered with Git Bash.
- Steps + observed results:
  1. Rendered the Summary step for Azure Managed 0.4.0 and confirmed it retains the Create PR link and identifies tag creation as the code-mirror trigger.
  2. Confirmed the prepare step pushes only the release branch without force and contains no tag mutation.
  3. Confirmed post-merge commands verify the merged commit/version and fail rather than overwrite an existing tag.
  4. Parsed `eng/ci/code-mirror.yml` and asserted its branch filter remains exactly `main` while its tag filter includes `*`.
  5. Confirmed the operator flow blocks official build/release until the internal tag resolves to the same `MERGED_COMMIT` and is locked.
- Evidence (optional): N/A

---

# Notes for reviewers
- `eng/ci/code-mirror.yml` extends `ci/code-mirror.yml@eng`. For tag refs, that shared engineering template dispatches to `ci/internal/code-mirror-tag.yml`, which compares the destination tag, deletes the stale ADO tag with the mirror identity when needed, pushes the canonical GitHub tag, and re-locks it. The new tag trigger activates that existing reconciliation path; it does not mutate the canonical GitHub tag.
- After this PR merges, reconcile the live mismatch by re-emitting/recreating the already-canonical GitHub `azuremanaged-v0.4.0` ref at `d447907899830e47b9af0eb3994e01e0a5d69beb`. That event should trigger the shared compare/delete/push/re-lock path under the mirror service identity. No separate ADO privilege intervention is expected if the shared template behaves as specified.

### CODE REVIEW

File: .github/workflows/prepare-release.yaml [hunk 1]
```diff
@@ -261,25 +261,22 @@ jobs:
           git config user.name "github-actions[bot]"
           git config user.email "github-actions[bot]@users.noreply.github.com"
           
-          # Create and checkout release branch (idempotent)
-          git checkout -B "$BRANCH_NAME"
+          # Create and checkout a new release branch.
+          git checkout -b "$BRANCH_NAME"
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply .github/workflows/prepare-release.yaml]: Addressed in 287cc23. Prepare Release now computes the target tag once and checks the exact `refs/tags/$TAG_NAME` on `origin` before changelog/version work. Existing tags fail immediately, and remote lookup errors also fail closed. I validated both an existing tag (`v0.3.0`) and an unused probe tag.

File: .github/workflows/prepare-release.yaml [hunk 2]
```diff
@@ -122,24 +122,44 @@ jobs:
           echo "new_version=$NEW_VERSION" >> $GITHUB_OUTPUT
           echo "New version: $NEW_VERSION"
 
+      - name: Verify target release tag is unused
+        id: target-tag
+        env:
+          NEW_VERSION: ${{ steps.calc-version.outputs.new_version }}
+          TAG_PREFIX: ${{ steps.resolve-package.outputs.tag_prefix }}
+        run: |
+          TAG_NAME="${TAG_PREFIX}${NEW_VERSION}"
```
[the remark this answers is not in the document]
Comment (YunchuWang) [reply .github/workflows/prepare-release.yaml:131]: No change needed here. Hyphenated step IDs with dot notation are the established syntax throughout this workflow (`resolve-package`, `get-current-version`, `calc-version`, `get-latest-tag`, `changelog-diff`, and `create-branch`), and the latest Prepare Release runs completed successfully with those expressions (for example, runs 30658622226 and 30657924353). The new `target-tag` ID follows that proven convention.
